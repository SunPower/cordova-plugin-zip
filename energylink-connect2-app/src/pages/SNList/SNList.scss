@import '~@sunpower/theme-dark/styles/variables';

.sn-modal{
  padding: 1rem;
  text-align: center;
  display: grid;
}

.snlist-blockui {
  display: flex;
  height:100%;

  .snlist {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    width: 100%;

    .has-no-border {
      border: 0;
    }

    .top-text {
      display: flex;
      flex-direction: column;
    }

    .sn-container {
      background-color: $silver-sand4;
      padding: $font-size-huge;
      margin: $font-size-medium 0;
      flex: 1 1 30rem;
      overflow-y: scroll;
      overflow-x: hidden;
      font-family: $font;
      letter-spacing: 0;
      border-radius: 5px;

      .line-scale-pulse-out-rapid {
        display: block;

        div {
          background-color: $sun;
        }
      }

      .sn-item {
        display: grid;
        grid-template-columns: 90% 1fr;
        text-align: left;
        padding-bottom: 5px;
        border-bottom: 2px solid;

        button {
          background: transparent;
          border: none;
        }
      }
    }
  }
}

