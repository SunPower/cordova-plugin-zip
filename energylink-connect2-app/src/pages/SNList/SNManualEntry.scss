.sn-manual-entry {
  .is-grouped-centered {
    align-items: center;
  }

  .control-barcode {
    flex-grow: 1;

    .field-barcode {
      width: 100%;
    }

    span.error {
      width: 235px;
      display: block;
      text-align: left;
    }
  }
  .control-add {
    flex-grow: 0;

    button {
      border: none;
      background-color: transparent;
    }
  }

}
