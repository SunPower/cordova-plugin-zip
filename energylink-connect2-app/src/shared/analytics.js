import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
