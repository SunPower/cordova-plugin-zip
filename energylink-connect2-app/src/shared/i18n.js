import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
