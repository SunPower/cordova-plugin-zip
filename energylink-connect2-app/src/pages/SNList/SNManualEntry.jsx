import React from 'react'

import TextField from '@sunpower/textfield'
import { useDispatch } from 'react-redux'
import { useI18n } from 'shared/i18n'
import { isValidSN } from 'shared/utils'
import { useField, useForm } from 'react-final-form-hooks'
import { ADD_PVS_SN } from 'state/actions/pvs'
import './SNManualEntry.scss'

const ManualEntryForm = () => {
  const t = useI18n()
  const dispatch = useDispatch()

  const { form, handleSubmit } = useForm({
    onSubmit: ({ barcode }) => {
      dispatch(ADD_PVS_SN({ serial_number: barcode }))
    },
    validate: values => {
      const errors = {}

      if (!isValidSN(values.barcode)) {
        errors.barcode = t('SN_FIELD_ERROR_REGEX')
      }
      return errors
    }
  })

  const fieldBarcode = useField('barcode', form)

  return (
    <form
      className="sn-manual-entry"
      onSubmit={event => {
        handleSubmit(event)
        form.reset()
      }}
    >
      <label htmlFor="barcode" className="has-text-white is-flex">
        {t('MI_SN_LABEL')}
      </label>

      <div className="field is-grouped is-grouped-centered">
        <div className="control">
          <TextField
            input={fieldBarcode.input}
            meta={fieldBarcode.meta}
            type="text"
            autoComplete="barcode"
            className="field-barcode"
          />
        </div>
        <div className="control">
          <button
            className="button is-primary is-uppercase pl-10 pr-10 pb-30 pt-10"
            type="submit"
          >
            {t('ADD')}
          </button>
        </div>
      </div>
    </form>
  )
}

export default ManualEntryForm
