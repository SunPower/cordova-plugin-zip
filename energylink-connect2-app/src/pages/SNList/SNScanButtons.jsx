import React from 'react'

import { useI18n } from 'shared/i18n'
import { either } from 'shared/utils'

const ScanButtons = ({ fetchingSN, onScanMore, countSN, canScanMore }) => {
  const t = useI18n()

  return (
    <div className="sn-buttons">
      {either(
        canScanMore,
        <button
          className="button half-button-padding is-secondary is-uppercase trigger-scan mr-10"
          onClick={onScanMore}
          disabled={fetchingSN}
        >
          {fetchingSN ? t('SCANNING_SN') : t('SCAN_MORE')}
        </button>
      )}

      <button
        className="button half-button-padding is-primary is-uppercase trigger-scan"
        onClick={countSN}
        disabled={fetchingSN}
      >
        {fetchingSN ? t('SCANNING_SN') : t('CONTINUE')}
      </button>
    </div>
  )
}

export default ScanButtons
