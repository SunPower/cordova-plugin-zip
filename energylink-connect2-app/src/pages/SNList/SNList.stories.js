import { action } from '@storybook/addon-actions/dist'
import { storiesOf } from '@storybook/react'
import React from 'react'
import { Provider } from 'react-redux'

import SNManualEntry from './SNManualEntry'

import SNList from '.'

import { configureStore } from 'state/store'

let initialState = {
  devices: {
    found: [{ SERIAL: '194355110333120', DEVICE_TYPE: 'Inverter' }]
  },
  global: {
    canAccessScandit: true
  },
  pvs: {
    serialNumbers: [
      {
        serial_number: 'E00121852014339',
        type: 'MI',
        model: 'Type E',
        bounding_box: {
          left: 504,
          top: 451,
          width: 1029,
          height: 86
        }
      },
      {
        serial_number: 'E00121852014348',
        type: 'MI',
        model: 'Type E',
        bounding_box: {
          left: 2655,
          top: 465,
          width: 1013,
          height: 75
        }
      },
      {
        serial_number: 'E00121929013075',
        type: 'MI',
        model: 'Type E',
        bounding_box: {
          left: 504,
          top: 848,
          width: 1009,
          height: 77
        }
      }
    ]
  },
  inventory: {
    bom: [
      { item: 'AC_MODULES', value: '0' },
      { item: 'DC_MODULES', value: '0' },
      { item: 'STRING_INVERTERS', value: '0' },
      { item: 'EXTERNAL_METERS', value: '0' },
      { item: 'ESS', value: '0' }
    ]
  }
}

storiesOf('Serial Number List Page', module)
  .add('Empty', () => {
    const { store } = configureStore(initialState)
    return (
      <div className="full-min-height pt-20 pb-20 pl-10 pr-10">
        <Provider store={store}>
          <SNList />
        </Provider>
      </div>
    )
  })
  .add('SNManualEntry', () => {
    const { store } = configureStore({})

    return (
      <div className="full-min-height pt-20 pb-20 pl-10 pr-10">
        <Provider store={store}>
          <SNManualEntry serialNumber="" callback={action('callback called')} />
        </Provider>
      </div>
    )
  })
