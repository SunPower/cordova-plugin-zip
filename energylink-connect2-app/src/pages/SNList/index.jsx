import React, { useCallback, useState } from 'react'
import { Loader } from 'components/Loader'
import useModal from 'hooks/useModal'
import { pathOr } from 'ramda'

import BlockUI from 'react-block-ui'

import 'react-block-ui/style.css'
import { useDispatch, useSelector } from 'react-redux'
import { useHistory, useLocation } from 'react-router-dom'
import paths from 'routes/paths'
import { useI18n } from 'shared/i18n'
import { PUSH_CANDIDATES_INIT } from 'state/actions/devices'
import { UPDATE_MI_COUNT } from 'state/actions/inventory'

import { REMOVE_SN, START_DISCOVERY_INIT } from 'state/actions/pvs'
import './SNList.scss'
import SNManualEntry from './SNManualEntry'
import SNScanButtons from './SNScanButtons'

function SNList() {
  const t = useI18n()
  const dispatch = useDispatch()
  const history = useHistory()
  const location = useLocation()
  const { isManualModeDefault = false } = pathOr({}, ['state'], location)

  const [isManualMode, setManualMode] = useState(isManualModeDefault)
  const { serialNumbers, fetchingSN } = useSelector(state => state.pvs)
  const { bom } = useSelector(state => state.inventory)

  const toggleManualMode = useCallback(() => setManualMode(!isManualMode), [
    isManualMode
  ])

  const modulesOnInventory = bom.filter(item => {
    return item.item === 'AC_MODULES'
  })
  const expectedMICount = modulesOnInventory[0].value
  const scannedMICount = serialNumbers.length

  const onScanMore = () => {
    history.push(paths.PROTECTED.SCAN_LABELS.path)
  }

  const handleRemoveSN = serialNumber => {
    dispatch(REMOVE_SN(serialNumber))
  }

  const createSnItem = serialNumber => {
    return (
      <div key={serialNumber} className="sn-item mt-10 mb-10 pb-5">
        <span className="is-uppercase has-text-weight-bold has-text-white">
          {serialNumber}
        </span>
        <button
          onClick={() => handleRemoveSN(serialNumber)}
          className="has-text-white is-size-6"
        >
          <i className="sp-trash" />
        </button>
      </div>
    )
  }

  const submitSN = () => {
    const snList = serialNumbers.map(device => {
      return { DEVICE_TYPE: 'Inverter', SERIAL: device.serial_number }
    })
    toggleSerialNumbersModal()
    dispatch(UPDATE_MI_COUNT(serialNumbers.length))
    dispatch(PUSH_CANDIDATES_INIT(snList))
    history.push(paths.PROTECTED.DEVICES.path)
  }

  const serialNumbersModalTemplate = text => {
    return (
      <div className="sn-modal">
        <span className="has-text-white mb-10">{text}</span>
        <div className="sn-buttons">
          <button
            className="button half-button-padding is-secondary trigger-scan mr-10"
            onClick={() => toggleSerialNumbersModal()}
          >
            {t('CANCEL')}
          </button>
          <button
            className="button half-button-padding is-primary trigger-scan"
            onClick={submitSN}
          >
            {t('CONTINUE')}
          </button>
        </div>
      </div>
    )
  }

  const getCountType = (scannedMICount, expectedMICount) => {
    const numExpected = parseInt(expectedMICount)
    const numScanned = parseInt(scannedMICount)

    if (parseInt(numScanned) === 0 && numExpected === 0) return 'MI_NO_COUNT'
    if (numScanned > numExpected) return 'MI_OVERCOUNT'
    return 'MI_UNDERCOUNT'
  }

  const serialNumbersModalContent = serialNumbersModalTemplate(
    t(
      getCountType(scannedMICount, expectedMICount),
      scannedMICount,
      expectedMICount
    )
  )

  const modalsTitle = (
    <span className="has-text-white has-text-weight-bold">
      {t('ATTENTION')}
    </span>
  )

  const startLegacyDiscovery = () => {
    dispatch(START_DISCOVERY_INIT({ Device: 'allplusmime' }))
    history.push(paths.PROTECTED.LEGACY_DISCOVERY.path)
  }

  const legacyDiscoveryModalContent = (
    <div className="sn-modal">
      <span className="has-text-white mb-10">
        {t('LEGACY_DISCOVERY_WARNING')}
      </span>
      <div className="sn-buttons">
        <button
          className="button half-button-padding is-secondary trigger-scan mr-10"
          onClick={() => toggleLegacyDiscoveryModal()}
        >
          {t('CANCEL')}
        </button>
        <button
          className="button half-button-padding is-primary trigger-scan"
          onClick={startLegacyDiscovery}
        >
          {t('CONTINUE')}
        </button>
      </div>
    </div>
  )

  const {
    modal: serialNumbersModal,
    toggleModal: toggleSerialNumbersModal
  } = useModal(serialNumbersModalContent, modalsTitle, false)

  const {
    modal: legacyDiscoveryModal,
    toggleModal: toggleLegacyDiscoveryModal
  } = useModal(legacyDiscoveryModalContent, modalsTitle, false)

  const countSN = () => {
    const parsedScannedMICount = parseInt(scannedMICount)
    const parsedExpectedMiCount = parseInt(expectedMICount)
    if (
      parsedScannedMICount !== parsedExpectedMiCount ||
      (parsedScannedMICount === 0 && parsedExpectedMiCount === 0)
    ) {
      toggleSerialNumbersModal()
    } else {
      submitSN()
      history.push(paths.PROTECTED.DEVICES.path)
    }
  }

  const serialNumbersList = serialNumbers
    ? serialNumbers
        .sort(function(a, b) {
          return a.serial_number > b.serial_number
        })
        .map(device => createSnItem(device.serial_number))
    : []

  return (
    <BlockUI
      tag="div"
      className="snlist-blockui"
      blocking={false}
      message={t('OPENING_CAMERA')}
    >
      {serialNumbersModal}
      {legacyDiscoveryModal}
      <div className="snlist is-vertical has-text-centered pl-10 pr-10">
        <div className="top-text">
          <span className="is-uppercase has-text-weight-bold">
            {t('SCAN_MI_LABELS')}
          </span>
          <span className="has-text-white">
            {serialNumbers.length > 0
              ? t('FOUND_SN', serialNumbers.length)
              : ''}
          </span>
        </div>
        <div className="sn-container">
          {serialNumbersList.length > 0 ? (
            serialNumbersList
          ) : (
            <span>{t('SCAN_HINT')}</span>
          )}
          {fetchingSN ? <Loader /> : ''}
        </div>

        <div className="sn-buttons">
          {isManualMode ? (
            <>
              <SNManualEntry toggleOpen={toggleManualMode} />
              <button
                onClick={toggleManualMode}
                className="button has-text-centered is-uppercase is-secondary has-no-border mr-40 pl-0 pr-0"
              >
                {t('BACK_TO_SCAN')}
              </button>
              <button
                onClick={toggleLegacyDiscoveryModal}
                className="button has-text-centered is-uppercase is-secondary has-no-border pl-0 pr-0"
              >
                {t('LEGACY_DISCOVERY')}
              </button>
            </>
          ) : (
            <>
              <SNScanButtons
                fetchingSN={fetchingSN}
                onScanMore={onScanMore}
                countSN={countSN}
              />
              <button
                onClick={toggleManualMode}
                className="button has-text-centered is-uppercase is-secondary has-no-border"
              >
                {t('SN_MANUAL_ENTRY')}
              </button>
            </>
          )}
        </div>
      </div>
    </BlockUI>
  )
}

export default SNList
