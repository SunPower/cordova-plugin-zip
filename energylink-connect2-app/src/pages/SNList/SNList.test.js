import React from 'react'
import * as reactRedux from 'react-redux'
import * as i18n from 'shared/i18n'
import * as ReactDOM from 'react-dom'
import SNList from '.'

describe('Serial Number List Page', () => {
  let dispatchMock

  let initialState = {
    pvs: {
      serialNumbers: [
        {
          serial_number: 'E00121852014339',
          type: 'MI',
          model: 'Type E',
          bounding_box: {
            left: 504,
            top: 451,
            width: 1029,
            height: 86
          }
        },
        {
          serial_number: 'E00121852014348',
          type: 'MI',
          model: 'Type E',
          bounding_box: {
            left: 2655,
            top: 465,
            width: 1013,
            height: 75
          }
        },
        {
          serial_number: 'E00121929013075',
          type: 'MI',
          model: 'Type E',
          bounding_box: {
            left: 504,
            top: 848,
            width: 1009,
            height: 77
          }
        }
      ]
    },
    inventory: {
      bom: [
        { item: 'AC_MODULES', value: '0' },
        { item: 'DC_MODULES', value: '0' },
        { item: 'STRING_INVERTERS', value: '0' },
        { item: 'EXTERNAL_METERS', value: '0' },
        { item: 'ESS', value: '0' }
      ]
    }
  }

  beforeAll(() => {
    ReactDOM.createPortal = jest.fn((element, node) => {
      return element
    })
  })

  beforeEach(() => {
    dispatchMock = jest.fn()
    jest.mock('react-router-dom', () => ({
      useHistory: () => ({
        history: {
          push: jest.fn()
        }
      })
    }))
    jest.spyOn(reactRedux, 'useDispatch').mockImplementation(() => dispatchMock)
    jest
      .spyOn(i18n, 'useI18n')
      .mockImplementation(path => (key, ...params) =>
        `${key.toUpperCase()} ${params.join('_')}`.trim()
      )
  })

  test('renders correctly', () => {
    const { component } = mountWithProvider(<SNList />)(initialState)
    expect(component).toMatchSnapshot()
  })
})
