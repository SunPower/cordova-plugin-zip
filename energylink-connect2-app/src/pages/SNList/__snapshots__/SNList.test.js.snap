// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Serial Number List Page renders correctly 1`] = `
<BlockUi
  blocking={false}
  loader={[Function]}
  message="OPENING_CAMERA"
  renderChildren={true}
  tag="div"
>
  <div
    className="snlist is-vertical has-text-centered pl-10 pr-10"
  >
    <div
      className="top-text"
    >
      <span
        className="is-uppercase has-text-weight-bold"
      >
        SCAN_MI_LABELS
      </span>
      <span
        className="has-text-white"
      >
        FOUND_SN 3
      </span>
    </div>
    <div
      className="sn-container"
    >
      <div
        className="sn-item mt-10 mb-10 pb-5"
        key="E00121852014339"
      >
        <span
          className="is-uppercase has-text-weight-bold has-text-white"
        >
          E00121852014339
        </span>
        <button
          className="has-text-white is-size-6"
          onClick={[Function]}
        >
          <i
            className="sp-trash"
          />
        </button>
      </div>
      <div
        className="sn-item mt-10 mb-10 pb-5"
        key="E00121852014348"
      >
        <span
          className="is-uppercase has-text-weight-bold has-text-white"
        >
          E00121852014348
        </span>
        <button
          className="has-text-white is-size-6"
          onClick={[Function]}
        >
          <i
            className="sp-trash"
          />
        </button>
      </div>
      <div
        className="sn-item mt-10 mb-10 pb-5"
        key="E00121929013075"
      >
        <span
          className="is-uppercase has-text-weight-bold has-text-white"
        >
          E00121929013075
        </span>
        <button
          className="has-text-white is-size-6"
          onClick={[Function]}
        >
          <i
            className="sp-trash"
          />
        </button>
      </div>
    </div>
    <button
      className="button is-primary trigger-scan"
      onClick={[Function]}
    >
      SCAN_MORE
    </button>
    <Link
      className="has-text-centered is-uppercase"
      to="/scan-labels"
    >
      SN_MANUAL_ENTRY
    </Link>
  </div>
</BlockUi>
`;
